packages/react/src/ReactLazy.js

```diff
@@ -52,14 +53,18 @@ function lazyInitializer<T>(payload: Payload<T>): T {
   if (payload._status === Uninitialized) {
     const ctor = payload._result;
     const thenable = ctor();
+    invariant(
+      thenable !== undefined && typeof thenable.then === 'function',
+      'lazy: Expects to receive an async function.',
+    );
     // Transition to the next state.
     const pending: PendingPayload = (payload: any);
     pending._status = Pending;
     pending._result = thenable;
     thenable.then(
```

Comment:
What I don't get is why doesn't this throw a proper error instead of `undefined`?

Reply:
I am assuming that 'invariant' is throwing an error, is it wrong?

Reply:
I think this problem has been resolved, as I believe the invariant is throwing an error properly.